packages/signature-controller/src/SignatureController.test.ts

```diff
@@ -343,321 +409,166 @@ describe('SignatureController', () => {
+        typedMessageManagerMock.addUnapprovedMessage,
       ).toHaveBeenCalledTimes(1);
-      expect(
-        typedMessageManagerMock.addUnapprovedMessageAsync,
-      ).toHaveBeenCalledWith(messageParamsMock, versionMock, requestMock);
-    });
-  });
-
-  describe.each([
-    [
-      'signMessage',
-      messageManagerMock,
-      () => signatureController.signMessage,
-      () => keyringControllerMock.signMessage,
-      'eth_sign',
-    ],
-    [
-      'signPersonalMessage',
-      personalMessageManagerMock,
-      () => signatureController.signPersonalMessage,
-      () => keyringControllerMock.signPersonalMessage,
-      'personal_sign',
-    ],
-    [
-      'signTypedMessage',
-      typedMessageManagerMock,
-      () => signatureController.signTypedMessage,
-      () => keyringControllerMock.signTypedMessage,
-      'eth_signTypedData',
-    ],
-  ])(
-    '%s',
-    (
-      signMethodName,
-      messageManager,
-      getSignatureControllerMethod,
-      getKeyringControllerMethod,
-      rpcMethodName,
-    ) => {
-      let signatureControllerMethod: (...args: any[]) => Promise<string>;
-      let keyringControllerMethod: jest.Mock;
-
-      // eslint-disable-next-line jest/no-duplicate-hooks
-      beforeEach(() => {
-        messageManager.approveMessage.mockResolvedValueOnce(messageParamsMock2);
-
-        (keyringControllerMock as any)[signMethodName].mockResolvedValueOnce(
-          signatureMock,
-        );
-
-        signatureControllerMethod =
-          getSignatureControllerMethod().bind(signatureController);
-        keyringControllerMethod = getKeyringControllerMethod();
-      });
-
-      it('approves message and signs', async () => {
-        await (signatureController as any)[signMethodName](messageParamsMock);
-
-        const keyringControllerExtraArgs =
-          // eslint-disable-next-line jest/no-if
-          signMethodName === 'signTypedMessage'
-            ? [{ version: messageParamsMock.version }]
-            : [];
-
-        expect(keyringControllerMethod).toHaveBeenCalledTimes(1);
-        expect(keyringControllerMethod).toHaveBeenCalledWith(
-          messageParamsMock2,
-          ...keyringControllerExtraArgs,
-        );
-
-        expect(messageManager.setMessageStatusSigned).toHaveBeenCalledTimes(1);
-        expect(messageManager.setMessageStatusSigned).toHaveBeenCalledWith(
-          messageParamsMock2.metamaskId,
-          signatureMock,
-        );
-      });
-
-      it('emits an event when the message is signed by the keyring', async () => {
-        const eventListener = jest.fn();
-
-        signatureController.hub.once(`${rpcMethodName}:signed`, eventListener);
-
-        await (signatureController as any)[signMethodName](messageParamsMock);
-
-        expect(eventListener).toHaveBeenCalledWith({
-          messageId: messageIdMock,
-          signature: signatureMock,
-        });
-      });
-
-      it('does not set as signed, messages with deferSetAsSigned', async () => {
```

Comment:
This file has changed a lot, so just wanted to verify we haven't lost any tests, specifically these MMI ones around the `deferSetAsSigned` flag?

Reply:
Added back one test especially for this case in `newUnsignedTypedMessage`